src/lib.rs

```diff
@@ -210,8 +211,9 @@ pub fn visual_runs(line: Range<usize>,
         // Look for the start of a sequence of consecutive runs of max_level or higher.
         let mut seq_start = 0;
         while seq_start < run_count {
-            if levels[runs[seq_start].start] < max_level {
+            if info.levels[runs[seq_start].start] < max_level {
                 seq_start += 1;
+                // FIXME: `continue` here?
```

Comment:
Yes, we should add `continue` here and also remove the following `if seq_start >= run_count { break }` which would be redundant.